install.py

```diff
@@ -217,32 +217,34 @@ def processTranslations(install, dst):
     stage = 'install'
 else:
     stage = 'removal'
-print '''Performing %s with:
-    destdir=%s
-    prefix=%s
-    sysconfdir=%s
-    examplesdir=%s
-    mandir=%s
-    pythonbin=%s''' % (stage, destdir, prefix, sysconfdir, examplesdir, mandir, pythonbin)
+print(f'''Performing {stage} with:
+    destdir={destdir}
+    prefix={prefix}
+    sysconfdir={sysconfdir}
+    examplesdir={examplesdir}
+    mandir={mandir}
+    pythonbin={pythonbin}''')
 
 # install files to prefix
 processFiles(install, os.path.join(destdir, prefix[1:]), 'src/usr/', {
-        'bin/diffuse': [ ("'../../etc/diffuserc'", repr(relpath(os.path.join(prefix, 'bin'), os.path.join(sysconfdir, 'diffuserc')))), ('/usr/bin/env python', pythonbin) ],
+        'bin/diffuse': [ (b"'../../etc/diffuserc'", repr(relpath(os.path.join(prefix, 'bin'), os.path.join(sysconfdir, 'diffuserc'))).encode()),
+            (b'/usr/bin/env python', pythonbin.encode()) ],
```

Comment:
why encode is needed for these strings especially?

Reply:
The main difference between py2 and py3 is that in py3 all strings are unicode, and it is not possible to mix strings and bytes. In processFiles we read (and write) files in binary mode:
f = open(s, 'rb')
c = f.read()
Here c has type bytes. This is the right thing to do, because some files (diffuse.png in src/usr/share/icons/hicolor/*/apps) are binary. But then we call
c = replace(c, v)
The second argument of replace, rules, is a list of 2-element tuples: find what and replace by what. In the old py2 versions, they were strings. This does not work in py3, both of them have to be bytes. Therefore I replaced
('/usr/bin/env python', pythonbin)
by
(b'/usr/bin/env python', pythonbin.encode())
The method encode converts the string pythonbin to bytes. If the encoding is not specified (the argument list is empty), it is supposed to be 'utf-8'. And this is exactly the right thing: all text files operated by install.py are in utf-8.